kernel/conversion.ml

```diff
@@ -54,8 +56,9 @@ let compare_stack_shape stk1 stk2 =
     | (_, Zapp l2::s2) -> compare_rec (bal-Array.length l2) stk1 s2
     | (Zproj _::s1, Zproj _::s2) ->
         Int.equal bal 0 && compare_rec 0 s1 s2
-    | (ZcaseT(_c1,_,_,_,_,_)::s1, ZcaseT(_c2,_,_,_,_,_)::s2) ->
-        Int.equal bal 0 (* && c1.ci_ind  = c2.ci_ind *) && compare_rec 0 s1 s2
+    | (ZcaseT(_c1,_,pms1,_,br1,_)::s1, ZcaseT(_c2,_,pms2,_,br2,_)::s2) ->
+        Int.equal bal 0 (* && c1.ci_ind  = c2.ci_ind *) && same_length pms1 pms2
```

Comment:
is there a reason not to compare the inductives? that would imply by typing that params and branches have the same size (it's been commented since 7c290a484d4167d91fbe2de84bba6931e2e2dd8f)

Reply:
I did it this way keeping in mind that at some point the canonical equality shall require an environment access.